avm/res/compute/gallery/image/main.bicep

```diff
@@ -256,3 +155,120 @@ type roleAssignmentType = {
   @sys.description('Optional. The Resource Id of the delegated managed identity resource.')
   delegatedManagedIdentityResourceId: string?
 }[]?
+
+@export()
+type imageType = {
+  @sys.description('Required. The resource name.')
+  @minLength(1)
+  @maxLength(80)
+  name: string
+
+  @sys.description('Optional. The location of the resource. Defaults to the gallery resource location.')
+  location: string?
+
+  @sys.description('Optional. Tags for all resources. Defaults to the tags of the gallery.')
+  @metadata({
+    example: '''
+  {
+      key1: 'value1'
+      key2: 'value2'
+  }
+  '''
+  })
+  tags: object?
+
+  @sys.description('Required. Describes the properties of a gallery image definition.')
+  properties: {
+    @sys.description('Optional. The architecture of the image. Applicable to OS disks only.')
+    architecture: 'x64' | 'Arm64'?
+
+    @sys.description('Optional. The description of this gallery image definition resource. This property is updatable.')
+    description: string?
+
+    @sys.description('Optional. Describes the disallowed disk types.')
+    disallowed: {
+      @sys.description('Required. A list of disk types.')
+      @metadata({ example: '''
+      [
+        'Standard_LRS'
+      ]''' })
+      diskTypes: string[]
+    }?
+
+    @sys.description('Optional. The end of life date of the gallery image definition. This property can be used for decommissioning purposes. This property is updatable.')
+    endOfLifeDate: string?
+
+    @sys.description('Optional. The Eula agreement for the gallery image definition.')
+    eula: string?
+
+    // features are skipped for now, as we prefer typed parameters
+
+    @sys.description('Optional. The hypervisor generation of the Virtual Machine. If this value is not specified, then it is determined by the securityType parameter. If the securityType parameter is specified, then the value of hyperVGeneration will be V2, else V1.')
+    hyperVGeneration: 'V1' | 'V2'?
+
+    @sys.description('Required. This is the gallery image definition identifier.')
+    identifier: {
+      @sys.description('Required. The name of the gallery image definition offer.')
+      publisher: string
+
+      @sys.description('Required. The name of the gallery image definition publisher.')
+      offer: string
+
+      @sys.description('Required. The name of the gallery image definition SKU.')
+      sku: string
+    }
+
+    @sys.description('Required. This property allows the user to specify whether the virtual machines created under this image are `Generalized` or `Specialized`.')
```

Comment:
Defaults in this and other parameters should not be part of the description as they're anyways part of the metadata (via the allowedSet)